src/ssl.c

```diff
@@ -8595,6 +8595,118 @@ int wolfSSL_check_private_key(const WOLFSSL* ssl)
 }
 
 #if defined(OPENSSL_ALL)
+unsigned int wolfSSL_X509_get_extension_flags(WOLFSSL_X509* x509)
+{
+    unsigned int flags = 0;
+
+    WOLFSSL_ENTER("wolfSSL_X509_get_extension_flags");
+
+    if (x509 == NULL) {
+        return flags;
+    }
+
+    if (x509->keyUsageSet) {
+        flags |= EXFLAG_KUSAGE;
+    }
+    if (x509->extKeyUsageSrc != NULL) {
+        flags |= EXFLAG_XKUSAGE;
+    }
+
+    WOLFSSL_LEAVE("wolfSSL_X509_get_extension_flags", flags);
+
+    return flags;
+}
+
+unsigned int wolfSSL_X509_get_key_usage(WOLFSSL_X509* x509)
+{
+    int ret = 0;
```

Comment:
My goal here is to be able to return -1 (aka 0xFFFFFFFF) as an error code. How would you propose I handle this?

Reply:
cast -1 to unsigned int?

Reply:
I don't see how that's superior to what I'm doing now, where I cast the return value to unsigned int.

Reply:
wolfSSL_X509_get_keyUsage() returns an unsigned int.
This function returns an unsigned int.

Reply:
Yes, and I'm returning an unsigned int via the existing cast, no? I'm not seeing the problem.

Reply:
Ah, is the concern that assigning ret to the result of `wolfSSL_X509_get_keyUsage` may cause a compiler warning in some cases, since ret is an int and that function returns unsigned int? Or is there a functional concern?

Reply:
The function is returning a bit mask which is an unsigned int value and makes no sense to be negative.

There are four places ret is used.
Three of them are as an unsigned int value:
 - assigned 0 for returning as an unsigned int
 - set from the return that is an unsigned int
 - returning ret as an unsigned int
The final usage is setting all the bits by assigning -1.

We are working with an unsigned int in this function but using an int because we don't want to cast -1 to an unsigned int.

Reply:
Either approach still seems equally fine, but I will change it so that we can move on.